File: examples/csharp/BasicUsage.cs
using System;
using System.Threading.Tasks;
using Sysmoon.SDK;

namespace SysmoonExamples
{
    class BasicUsage
    {
        static async Task Main(string[] args)
        {
            var client = new SysmoonClient(new SysmoonConfig
            {
                ApiUrl = "http://localhost:3001"
            });

            try
            {
                // Register a new system
                Console.WriteLine("Registering system...");
                var registration = await client.RegisterAsync(
                    "Example C# App",
                    "Demo application for Sysmoon"
                );

                Console.WriteLine("✓ System registered successfully!");
                Console.WriteLine($"  System ID: {registration.SystemId}");
                Console.WriteLine($"  API Key: {registration.ApiKey}");
                Console.WriteLine("  ⚠️  Save the API key - you won't see it again!\n");

                // Send some example events
                Console.WriteLine("Sending events...");

                await client.SendEventAsync(new EventData
                {
                    EventType = "app.started",
                    Payload = new
                    {
                        Version = "1.0.0",
                        Environment = "development",
                        Timestamp = DateTime.UtcNow
                    },
                    Severity = "info"
                });
                Console.WriteLine("✓ Sent: app.started");

                await client.SendEventAsync(new EventData
                {
                    EventType = "user.login",
                    Payload = new
                    {
                        UserId = "user123",
                        Username = "johndoe",
                        Ip = "192.168.1.100"
                    },
                    Severity = "info"
                });
                Console.WriteLine("✓ Sent: user.login");

                // Send batch events
                await client.SendEventsAsync(new[]
                {
                    new EventData
                    {
                        EventType = "database.query",
                        Payload = new { Query = "SELECT * FROM users", Duration = 45 },
                        Severity = "info"
                    },
                    new EventData
                    {
                        EventType = "api.request",
                        Payload = new { Endpoint = "/api/users", Method = "GET", Status = 200 },
                        Severity = "info"
                    }
                });
                Console.WriteLine("✓ Sent: 2 batch events");

                // Demonstrate error event
                await client.SendEventAsync(new EventData
                {
                    EventType = "error.database",
                    Payload = new
                    {
                        Error = "Connection timeout",
                        Query = "SELECT * FROM orders",
                        Duration = 5000
                    },
                    Severity = "error"
                });
                Console.WriteLine("✓ Sent: error.database\n");

                Console.WriteLine("All events sent successfully!");
                Console.WriteLine("Check the dashboard at http://localhost:3000 to see them.");
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine($"Error: {ex.Message}");
                Environment.Exit(1);
            }
            finally
            {
                client.Dispose();
            }
        }
    }
}

File: examples/nodejs/basic-usage.js
// Example: Basic event sending with JavaScript SDK
import SysmoonClient from '@sysmoon/sdk-js';

const client = new SysmoonClient({
  apiUrl: 'http://localhost:3001',
});

async function main() {
  try {
    // Register a new system
    console.log('Registering system...');
    const registration = await client.register(
      'Example Node.js App',
      'Demo application for Sysmoon'
    );
    
    console.log('✓ System registered successfully!');
    console.log(`  System ID: ${registration.systemId}`);
    console.log(`  API Key: ${registration.apiKey}`);
    console.log('  ⚠️  Save the API key - you won\'t see it again!\n');
    
    // Send some example events
    console.log('Sending events...');
    
    await client.sendEvent({
      eventType: 'app.started',
      payload: {
        version: '1.0.0',
        environment: 'development',
        timestamp: new Date().toISOString(),
      },
      severity: 'info',
    });
    console.log('✓ Sent: app.started');
    
    await client.sendEvent({
      eventType: 'user.login',
      payload: {
        userId: 'user123',
        username: 'johndoe',
        ip: '192.168.1.100',
      },
      severity: 'info',
    });
    console.log('✓ Sent: user.login');
    
    // Send batch events
    await client.sendEvents([
      {
        eventType: 'database.query',
        payload: { query: 'SELECT * FROM users', duration: 45 },
        severity: 'info',
      },
      {
        eventType: 'api.request',
        payload: { endpoint: '/api/users', method: 'GET', status: 200 },
        severity: 'info',
      },
    ]);
    console.log('✓ Sent: 2 batch events');
    
    // Demonstrate error event
    await client.sendEvent({
      eventType: 'error.database',
      payload: {
        error: 'Connection timeout',
        query: 'SELECT * FROM orders',
        duration: 5000,
      },
      severity: 'error',
    });
    console.log('✓ Sent: error.database\n');
    
    console.log('All events sent successfully!');
    console.log('Check the dashboard at http://localhost:3000 to see them.');
    
  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  }
}

main();
